src/test/java/uk/gov/hmcts/reform/professionalapi/utils/RefDataUtilTest.java

```diff
@@ -249,4 +251,91 @@ public void test_filterUsersByStatusWhereNoUsersFoundThrowsResourceNotFoundExcep
 
         RefDataUtil.filterUsersByStatus(responseEntity, "Active");
     }
+
+    @Test
+    public void test_shouldGenerateResponseEntityWithHeaderFromPage() {
+        HttpHeaders responseHeader = new HttpHeaders();
+        responseHeader.add("fakeHeader", "fakeValue");
+
+        ResponseEntity responseEntityMock = mock(ResponseEntity.class);
+        when(responseEntityMock.getHeaders()).thenReturn(responseHeader);
+
+        Page pageMock = mock(Page.class);
+        when(pageMock.getTotalElements()).thenReturn(1L);
+        when(pageMock.getTotalPages()).thenReturn(2);
+
+        Pageable pageableMock = mock(Pageable.class);
+        when(pageableMock.getPageNumber()).thenReturn(0);
+        when(pageableMock.getPageSize()).thenReturn(2);
+
+        HttpHeaders httpHeaders = RefDataUtil.generateResponseEntityWithPaginationHeader(pageableMock, pageMock, responseEntityMock);
+
+        assertThat(httpHeaders.containsKey("fakeHeader")).isTrue();
+        assertThat(httpHeaders.containsKey("paginationInfo")).isTrue();
+
+        //MultiValueMap<String, String> multiValueMap = new LinkedMultiValueMap<>();
+        //HttpHeaders httpHeadersMock = mock(HttpHeaders.class);
+
+        //verify(httpHeadersMock, times(1)).addAll(multiValueMap);
+        //verify(httpHeadersMock, times(1)).add(any(String.class), any(String.class));
+    }
+
+    @Test
+    public void test_shouldGenerateResponseEntityWithHeaderFromPageWhenResponseEntityIsNull() {
+        Page pageMock = mock(Page.class);
+        when(pageMock.getTotalElements()).thenReturn(1L);
+        when(pageMock.getTotalPages()).thenReturn(2);
+
+        Pageable pageableMock = mock(Pageable.class);
+        when(pageableMock.getPageNumber()).thenReturn(0);
+        when(pageableMock.getPageSize()).thenReturn(2);
+
+        HttpHeaders httpHeaders = RefDataUtil.generateResponseEntityWithPaginationHeader(pageableMock, pageMock, null);
+
+        HttpHeaders httpHeadersMock = mock(HttpHeaders.class);
+
+        assertThat(httpHeaders.containsKey("paginationInfo")).isTrue();
+        //verify(httpHeadersMock, times(1)).add(any(String.class), any(String.class));

```

Comment:
Please add AT LEAST ONE verify LOC and/or assertion if you are going to have a new test.